eng/doc/fips/README.md

```diff
@@ -33,69 +31,165 @@ On Linux, the fork uses [OpenSSL](https://www.openssl.org/) through the [go-cryp
+  - [`requirefips` build tag](#build-option-to-require-fips-mode)
+  - [`GOEXPERIMENT` `allowcryptofallback`](#build-option-to-allow-go-standard-library-crypto-fallback)
+  - [`import _ "crypto/tls/fipsonly"` source change](#tls-with-fips-approved-settings)
+- Runtime configuration:
+  - [`GOFIPS` environment variable](#usage-runtime)
+  - (OpenSSL backend) [`GO_OPENSSL_VERSION_OVERRIDE` environment variable](#runtime-openssl-version-override)
+  - (OpenSSL backend) [`/proc/sys/crypto/fips_enabled` file containing `1`](#linux-fips-mode-openssl)
+  - (CNG backend) [Windows registry `HKLM\SYSTEM\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy` dword value `Enabled` set to `1`](#windows-fips-mode-cng)
+
+## Usage: Common configurations
+
+There are typically two goals that lead to this document. Creating a FIPS compliant app is one. The other is to comply with internal Microsoft crypto policies that have been set for Go. This table summarizes common configurations and how suitable each one is for these goals.
+
+> This section assumes the use of Microsoft Go 1.21 or later. As of writing, Go 1.21 is scheduled for release in the second week of August 2023.
+>
+> 1.21 introduces `systemcrypto`, `requirefips`, and improvements to crypto fallback behavior. The Usage sections go into more detail about the differences between 1.19/1.20 and 1.21 in context. See also [the Go 1.21 changelog](#go-121).
+
+| Build-time config | Runtime config | Internal Microsoft crypto policy | FIPS behavior |
+| --- | --- | --- | --- |
+| Default | Default | Not compliant | Crypto usage is not FIPS compliant. |
+| `GOEXPERIMENT=systemcrypto` | Default | Compliant | Can be used to create a compliant app. FIPS mode is automatically enabled if it is configured systemwide or if `GOFIPS` is enabled at some point in the future. Flexible. |
+| `GOEXPERIMENT=systemcrypto` | `GOFIPS=1` | Compliant | Can be used to create a compliant app. The app either enables FIPS mode or ensures it is already enabled. Otherwise, the app panics. |
+| `GOEXPERIMENT=systemcrypto` | `GOFIPS=0` | Compliant | Crypto usage is not FIPS compliant. The app attempts to disable FIPS mode and panics if it isn't possible. |
+| `GOEXPERIMENT=systemcrypto` | `GO_OPENSSL_VERSION_OVERRIDE=1.1.1k-fips` | Compliant | Can be used to create a compliant app. If the app is built for Linux, `systemcrypto` chooses `opensslcrypto`, and the environment variable causes it to load `libcrypto.so.1.1.1k-fips` instead of using the automatic search behavior. This environment variable has no effect with `cngcrypto`. |
+| `GOEXPERIMENT=systemcrypto` and `-tags=requirefips` | Default | Compliant | Can be used to create a compliant app. The behavior is the same as `GOFIPS=1`, but no runtime configuration is necessary. See [the `requirefips` section](#build-option-to-require-fips-mode) for more information on when this "locked-in" approach may be useful rather than the flexible approach. |
```

Comment:
Think we should list the `allowcryptofallback` case here. Customers found it before, they'll find it again, best to acknowledge the combo exists and call it out as explicitly not compliant.

Reply:
It actually is compliant if used correctly, it just introduces risk. In any case, sure, adding.

Reply:
I put it into its own table at the end of this section instead to provide a little more context.